components/dashboard/src/provider-utils.tsx

```diff
@@ -94,9 +95,12 @@ function openModalWindow(url: string) {
     );
 }
 
-function attachMessageListener({ onSuccess, onError }: WindowMessageHandler) {
+function attachMessageListener(successKey: string, { onSuccess, onError }: WindowMessageHandler) {
     const eventListener = (event: MessageEvent) => {
-        // todo: check event.origin
+        // TODO: check w/ why we didn't do this already
```

Comment:
IMO nobody was aware that we need to. :shrug:

Reply:
> or just that we didn't get to it

It is a tedious procedure to provoke/test this one. After some consideration of security impact at the time, I didn't continue, but left a comment in place. The thinking was more like, if an attacker would make it happen to post window messages from a different origin, the worst thing was a awkward experience. Tampering with user's data was not expected, as the effect of the called handler function was to reload or refresh the context.

Reply:
Yah, as long as the callback handlers that fire can't cause any harm aside from perhaps putting the UI in a weird state then you're right, it isn't a huge deal. 

The check seems safe to me since we are always the one's controlling the auth popup flow, so it'll come from the same origin.